File: lobwebapp-core/src/main/java/com/espindola/lobwebapp/service/contract/StockService.java
package com.espindola.lobwebapp.service.contract;

import com.espindola.lobwebapp.domain.Stock;
import com.espindola.lobwebapp.service.contract.base.EntityService;

public interface StockService extends EntityService<Stock> {
	Stock findByProductId(Long productId);
}

File: lobwebapp-html/src/script/service/contract/StockService.ts
///<reference path="../../reference.d.ts"/>

module d.service.contract {
    export interface StockService extends d.service.contract.base.EntityService<domain.Stock> {

    	listUnit(
            successCallback: (data: string[], status: number, headers: (headerName: string) => string, config: ng.IRequestConfig) => any,
            errorCallback: (data: domain.util.MessageResponse, status: number, headers: (headerName: string) => string, config: ng.IRequestConfig) => any
            ): void;
    }
}